authz.go

```diff
@@ -225,6 +226,20 @@ func (a *authorizer) authorize(ctx context.Context, sr authz.SubjectAccessReview
 			break
 		}
 	}
+
+	// if cache and dry run are enabled, error need to be added to log if authorize result from cache and authorize result from zts / zms are different.
+	if a.AuthorizationConfig.Config.UseCache {
+		// decision is nil, meaning it is not authorized through cache, but zts/zms authorized
+		if decision == nil && granted {
```

Comment:
Can we remove the granted check and log it if it's either true or false?

Reply:
if you refer to this function: https://github.com/yahoo/k8s-athenz-webhook/blob/master/authz.go#L125. if cache doesn't authorize the check, it will return nil. So this if statement is for the mismatch between unauthorized from cache and authorized from zts/zms.

Reply:
Yep, what I meant was that if the granted value is true or false and decision is nil we should log regardless as we want to see what zts / zms returned even if it's false.